File: ColegioTrenerAndroid/app/src/main/java/com/jjmf/colegiotrenerandroid/data/dto/CitaInformeDto.kt
package com.jjmf.colegiotrenerandroid.data.dto

import com.google.gson.annotations.SerializedName
import com.jjmf.colegiotrenerandroid.domain.model.CitaInforme

data class CitaInformeDto(
    @SerializedName("nalumno") val nalumno: String?,
    @SerializedName("clase") val clase: String?,
    @SerializedName("fechacita") val fechacita: String?,
    @SerializedName("horario") val horario: String?,
    @SerializedName("observa") val observa: String?
) {
    fun toDomain(): CitaInforme {
        return CitaInforme(
            nalumno = nalumno?.trim(),
            clase = clase?.trim(),
            fechacita = fechacita?.trim(),
            horario = horario?.trim(),
            observa = observa?.trim()
        )
    }
}

File: ColegioTrenerSwift/ColegioTrenerSwift/Network/Dto/CitaInformeDto.swift
//
//  CitaInformeDto.swift
//  ColegioTrenerSwift
//
//  Created by Joel Maldonado on 1/05/24.
//

import SwiftUI

struct CitaInformeDto: Codable {
    let nalumno: String?
    let clase: String?
    let fechacita: String?
    let horario: String?
    let observa: String?
    let linkinforme: String?
    
    func toDomain() -> CitaInforme {
        let link = self.linkinforme.flatMap { URL(string: $0) }
        return CitaInforme(
            nalumno: nalumno?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            clase: clase?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            fechacita: fechacita?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            horario: horario?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            observa: observa?.trimmingCharacters(in: .whitespacesAndNewlines) ?? "",
            linkinforme: link
        )
    }
}
